server/index.js

```diff
@@ -237,6 +243,31 @@ export default class Server {
     }
   }
 
+  async readBuildId () {
+    const buildIdPath = join(this.dir, '.next', 'BUILD_ID')
+    try {
+      const buildId = await fs.readFile(buildIdPath, 'utf8')
+      return buildId.trim()
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        return '-'
```

Comment:
I'm quite not sure how to deal it with HMR and others. That's why I didn't do it.
Anyway, https://github.com/zeit/next.js/issues/786 has no relation to this.